src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -3288,7 +3288,8 @@ static void _svgLoaderParserXmlClose(SvgLoaderData* loader, const char* content,
     for (unsigned int i = 0; i < sizeof(graphicsTags) / sizeof(graphicsTags[0]); i++) {
         if (!strncmp(tagName, graphicsTags[i].tag, sz)) {
             loader->currentGraphicsNode = nullptr;
-            loader->stack.pop();
+            if (!strncmp(tagName, "text", 4)) loader->openedTag = OpenedTagType::Other;
```

Comment:
There may be cases like this.
`<text> text <some tags.../>  txt </text> ` https://www.w3.org/TR/SVG2/text.html#TextElement
Since text also acts like a group, we can push it onto the stack.

```diff
--- a/src/loaders/svg/tvgSvgLoader.cpp
+++ b/src/loaders/svg/tvgSvgLoader.cpp
@@ -3360,7 +3360,10 @@ static void _svgLoaderParserXmlOpen(SvgLoaderData* loader, const char* content,
         else parent = loader->doc;
         node = method(loader, parent, attrs, attrsLength, simpleXmlParseAttributes);
         if (node && !empty) {
-            if (!strcmp(tagName, "text")) loader->openedTag = OpenedTagType::Text;
+            if (!strcmp(tagName, "text")) {
+                loader->openedTag = OpenedTagType::Text;
+                loader->stack.push(node);
+            }
             else {
                 auto defs = _createDefsNode(loader, nullptr, nullptr, 0, nullptr);
                 loader->stack.push(defs);

```

When I tested with the icon_looks_left_text.svg sample, the icon is displayed fine, but the text is not displayed, so I don't know what other problem is left.

Reply:
IIRC, embedding the required fonts isn't yet implemented. You currently need to load an external font and give it the same name (family), e.g.:

```cpp
     // font
         // ---- Load from memory. ----
         {
            ifstream file(font_path, ios::binary);
            if (!file.is_open())
            {
               return false;
            }
            file.seekg(0, std::ios::end);
            std::streampos size = file.tellg();
            file.seekg(0, std::ios::beg);

            char *data = (char *)malloc(size);
            if (data && file.read(data, size))
            {
               tvg::Result result = tvg::Text::load(font_name.c_str(), data, size, "ttf", true);
...
```

Reply:
> IIRC, embedding the required fonts isn't yet implemented. You currently need to load an external font and give it the same name (family), e.g.:

Thanks :) I loaded the fontfamily and tested it and I checked the text is displayed.